qutip/core/data/make.py

```diff
@@ -0,0 +1,108 @@
+from .dispatch import Dispatcher as _Dispatcher
+import numpy as np
+from scipy.sparse import diags as scipy_diags
+from .dense import Dense
+from .csr import CSR
+from . import csr, dense, CSR, Dense
+from numbers import Number
+
+__all__ = ['diag',
+           'one_element_csr', 'one_element_dense', 'one_element']
+
+
+def diag_signature(diagonals, offsets=None, shape=None):
+    """
```

Comment:
I can't remember where or how, but I'm pretty sure somewhere in one of the other files I found a "cleaner" way of doing this without leaking a function name, perhaps using `inspect`?